docs/concepts/native-interop.rst

```diff
@@ -0,0 +1,254 @@
+Native interoperability
+=======================
+
+In this document we will dive a little bit deeper into all three ways of doing 
+"native interoperability" that are available on the .NET platform. 
+
+There are a couple of reasons why you would want to call into native code:
+
+* Operating Systems come with a large volume of APIs that are 
+  not present in the managed class libraries. A prime example for this would be 
+  access to hardware or operating system management functions.
+* Communicating with components that are written in unmanaged languages usually 
+  means using P/Invoke. This covers components written in other managed languages,
+  such as Java for instance.
+* On Windows, most of the software that gets installed, such as Microsoft Office
+  suite, registers COM components that represent their programs and allow developers 
+  to automate them or use them. This also requires native interoperability.
+
+There are certainly more situations where it is apt to use native code, especially 
+those that are specific for a given developer's scenario. 
+
+Platform Invoke (P/Invoke)
+--------------------------
+P/Invoke is a technology that allows you to access structs, callbacks and functions
```

Comment:
I don't think it is an issue of "accessing" these things, but using structs and friends to call native APIs.

P/Invoke does have a set of attributes. You can also pass things like bool, byte and int. I don't really consider those the P/Invoke API, per se.

The last sentence either ends early or isn't really helpful. It sounds like you need to better describe what constitutes the P/Invoke API.